netcreateorg/netcreate-2018
# Feature: Focus

NOTE: Merge #264 First!

This is a rough first pass at the focus feature.


It made most sense as another temporary "View" modification tied to Filters, so I added it to the panel formerly known as "Filters" but is now called "Views".

# To Test
1. Quit and restart the server.
2. Select a node
3. Click on "Views" panel
4. Click on the "Focus" tab
5. The graph should show your selected node and one degree of separation from that node
6. Use the stepper buttons to move the range up and down.
7. You should be able to increase it to any number
8. You should not be able to able to decrease it below 1
9. You should be able to use backspace to delete the range value and type in a new value
10. If you type in a 0 or negative value, the input field will remain blank
11. Try different range values to make sure the selection range is correct.
12. Open the Nodes Table => The nodes listed in the table should match the visible nodes, any nodes not graph should be removed.
13. Open the Edges Table => The edges should match
14. Clicking "Clear Filters" should restore all the nodes

There may very well be some algorithm issues with the way I'm crawling the network, so please verify that it does what you expect.
There may also be some weird interim states if you're going back and forth between the tables and the filters.


I've also attached a sample project for testing.  Just unzip and put it in `netcreate-2023/build/runtime`
[focus.zip](https://github.com/netcreateorg/netcreate-2018/files/11630271/focus.zip)
